readthedocs/doc_builder/base.py

```diff
@@ -84,6 +86,13 @@ def clean(self, **__):
     def docs_dir(self, docs_dir=None, **__):
         """Handle creating a custom docs_dir if it doesn't exist."""
         checkout_path = self.project.checkout_path(self.version.slug)
+        if docs_dir:
+            possible_path = os.path.join(checkout_path, docs_dir)
+            # if user puts an invalid `docs_dir` path raise Exception
+            if not os.path.exists(possible_path):
+                raise MkDocsYAMLParseError(
+                    MkDocsYAMLParseError.INVALID_DOCS_DIR_PATH,
+                )
```

Comment:
We shouldn't put special cases in the base class, the base class should know nothing about mkdocs.

Reply:
Got it :)

Reply:
@stsewd Updated the PR